cpp/src/Slice/Parser.cpp

```diff
@@ -563,27 +564,49 @@ Slice::Contained::name() const
 string
 Slice::Contained::scoped() const
 {
-    return _scoped;
+    return scope() + name();
 }
 
 string
 Slice::Contained::scope() const
 {
-    string::size_type idx = _scoped.rfind("::");
-    assert(idx != string::npos);
-    return string(_scoped, 0, idx + 2);
+    string scoped;
+    if (auto container = dynamic_pointer_cast<Contained>(_container))
+    {
+        scoped = container->scoped();
+    }
+    return scoped + "::";
 }
 
 string
-Slice::Contained::flattenedScope() const
+Slice::Contained::mappedName() const
 {
-    string s = scope();
-    string::size_type pos = 0;
-    while ((pos = s.find("::", pos)) != string::npos)
+    // First check if any 'xxx:identifier' has been applied to this element.
+    // If so, we return that instead of the element's Slice identifier.
+    static const string metadata = string(_unit->languagePrefix()) + ":identifier";
```

Comment:
I find this `static const` dubious. Yes, all the units in your program will most likely have the same language prefix, but at the same time, you made language prefix a per-unit field, not a global field.

Reply:
I find it very unlikely that this would actually cause any problems.
But, I removed the `static` anyways.

So we'll re-compute this string every time the function is called.
Thankfully the compilers are already pretty fast, so it's fine I guess.